packages/runtime/src/useCases/consumption/attributes/ChangeDefaultRepositoryAttribute.ts

```diff
@@ -0,0 +1,42 @@
+import { Result } from "@js-soft/ts-utils";
+import { AttributesController, LocalAttribute } from "@nmshd/consumption";
+import { AccountController, CoreId } from "@nmshd/transport";
+import { Inject } from "typescript-ioc";
+import { LocalAttributeDTO } from "../../../types";
+import { AttributeIdString, RuntimeErrors, SchemaRepository, SchemaValidator, UseCase } from "../../common";
+import { AttributeMapper } from "./AttributeMapper";
+
+export interface ChangeDefaultRepositoryAttributeRequest {
+    attributeId: AttributeIdString;
+}
+
+class Validator extends SchemaValidator<ChangeDefaultRepositoryAttributeRequest> {
+    public constructor(@Inject schemaRepository: SchemaRepository) {
+        super(schemaRepository.getSchema("ChangeDefaultRepositoryAttributeRequest"));
+    }
+}
+
+export class ChangeDefaultRepositoryAttributeUseCase extends UseCase<ChangeDefaultRepositoryAttributeRequest, LocalAttributeDTO> {
+    public constructor(
+        @Inject private readonly attributesController: AttributesController,
+        @Inject private readonly accountController: AccountController,
+        @Inject validator: Validator
+    ) {
+        super(validator);
+    }
+
+    protected async executeInternal(request: ChangeDefaultRepositoryAttributeRequest): Promise<Result<LocalAttributeDTO>> {
+        const repositoryAttribute = await this.attributesController.getLocalAttribute(CoreId.from(request.attributeId));
+        if (!repositoryAttribute) return Result.fail(RuntimeErrors.general.recordNotFound(LocalAttribute));
+
+        if (!repositoryAttribute.isRepositoryAttribute(this.accountController.identity.address)) {
+            return Result.fail(RuntimeErrors.attributes.isNotRepositoryAttribute(CoreId.from(request.attributeId)));
+        }
+
+        const defaultRepositoryAttribute = await this.attributesController.changeDefaultRepositoryAttribute(repositoryAttribute);
```

Comment:
Should we allow the user to manually set a RepositoryAttribute as default that already has a successor? Maybe this wouldn't be great for UX.

Reply:
Currently, the App doesn't show the predecessors at all. Thus, for now I'll prohibit to set outdated versions as default. If at a later point in time we'll expose the predecessors, we can think about adjusting the validation here, as well.